x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/api_key_configuration.tsx

```diff
@@ -96,17 +98,29 @@ export const ApiKeyConfig: React.FC<{ hasApiKey: boolean; indexName: string }> =
       {isModalVisible && <ConfirmModal onCancel={onCancel} onConfirm={onConfirm} />}
       <EuiFlexItem>
         <EuiText size="s">
-          {i18n.translate(
-            'xpack.enterpriseSearch.content.indices.configurationConnector.apiKey.description',
-            {
-              defaultMessage:
-                'First, generate an Elasticsearch API key. This {apiKeyName} key will enable read and write permissions for the connector to index documents to the created {indexName} index. Save the key in a safe place, as you will need it to configure your connector.',
-              values: {
-                apiKeyName: `${indexName}-connector`,
-                indexName,
-              },
-            }
-          )}
+          {!isNative
+            ? i18n.translate(
+                'xpack.enterpriseSearch.content.indices.configurationConnector.apiKey.description',
+                {
+                  defaultMessage:
+                    'First, generate an Elasticsearch API key. This {apiKeyName} key will enable read and write permissions for the connector to index documents to the created {indexName} index. Save the key in a safe place, as you will need it to configure your connector.',
+                  values: {
+                    apiKeyName: `${indexName}-connector`,
+                    indexName,
+                  },
+                }
+              )
+            : i18n.translate(
+                'xpack.enterpriseSearch.content.indices.configurationConnector.nativeConnector.apiKey.description',
+                {
+                  defaultMessage:
+                    'This native connector\'s API key {apiKeyName} is managed internally by Elasticsearch, and is used for the connector to index documents to the created {indexName} index. There\'s no need to do anything here, but if you need to rollover your API key you can do so by clicking "Generate API key".',
```

Comment:
nitpick: the writing tone here is is a little too casual imo. Also ideally we shouldn't be escaping apostrophes but using a backtick-based string.

Reply:
Is it the `There's no need to do anything here` part that is too casual?
How does this sound:

`This native connector's API key {apiKeyName} is managed internally by Elasticsearch, and is used for the connector to index documents to the created {indexName} index. If you need to rollover your API key you can do so by clicking "Generate API key".`

Also I would like to link documentation here once we've written some, if that helps.

Reply:
Little copyedit for concision:

```This native connector's API key {apiKeyName} is managed internally by Elasticsearch. The connector uses this API key to index documents into the {indexName} index. To rollover your API key, click "Generate API key".```